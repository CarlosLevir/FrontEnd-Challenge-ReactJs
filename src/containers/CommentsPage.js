import React, { Component } from 'react';
import { connect } from 'react-redux';
import CommentsCard from '../presentations/CommentsCard';

const styles = {
  container: {
    display: 'flex',
    flexWrap: 'wrap',
    alignItems: 'center',
    justifyContent: 'center'
  }
};

class CommentsPage extends Component {
  renderItem = (comment, key) => <CommentsCard comment={comment} key={key} />;

  render() {
    const { comments } = this.props;
    const renderedComments = comments.map((comment, key) => this.renderItem(comment, key));
    return <div style={styles.container}>{renderedComments}</div>;
  }
}

const mapStateToProps = state => ({
  comments: state.comments.comments
});

export default connect(mapStateToProps)(CommentsPage);
