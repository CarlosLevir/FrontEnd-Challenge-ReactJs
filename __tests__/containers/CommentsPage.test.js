import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import CommentsPage from '../../src/containers/CommentsPage';

const mockStore = configureStore();

const initialState = {
  comments: [
    {
      name: 'Lorem Ipsum',
      email: 'lorem@ipsum',
      body: 'Lorem Ipsum'
    },
    {
      name: 'Lorem Ipsum',
      email: 'lorem@ipsum',
      body: 'Lorem Ipsum'
    }
  ]
};

const context = React.createContext(initialState);
const store = mockStore(context);

configure({ adapter: new Adapter() });

describe('CommentsPage', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <CommentsPage />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
