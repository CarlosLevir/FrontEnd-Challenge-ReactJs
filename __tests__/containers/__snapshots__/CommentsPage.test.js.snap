// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsPage renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(CommentsPage) />
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Connect(CommentsPage) />,
      "value": Object {
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeState": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": Object {
            "comments": Array [
              Object {
                "body": "Lorem Ipsum",
                "email": "lorem@ipsum",
                "name": "Lorem Ipsum",
              },
              Object {
                "body": "Lorem Ipsum",
                "email": "lorem@ipsum",
                "name": "Lorem Ipsum",
              },
            ],
          },
          "_currentValue2": Object {
            "comments": Array [
              Object {
                "body": "Lorem Ipsum",
                "email": "lorem@ipsum",
                "name": "Lorem Ipsum",
              },
              Object {
                "body": "Lorem Ipsum",
                "email": "lorem@ipsum",
                "name": "Lorem Ipsum",
              },
            ],
          },
          "_threadCount": 0,
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Connect(CommentsPage) />,
        "value": Object {
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "storeState": Object {
            "$$typeof": Symbol(react.context),
            "Consumer": Object {
              "$$typeof": Symbol(react.context),
              "_calculateChangedBits": null,
              "_context": [Circular],
            },
            "Provider": Object {
              "$$typeof": Symbol(react.provider),
              "_context": [Circular],
            },
            "_calculateChangedBits": null,
            "_currentRenderer": null,
            "_currentRenderer2": null,
            "_currentValue": Object {
              "comments": Array [
                Object {
                  "body": "Lorem Ipsum",
                  "email": "lorem@ipsum",
                  "name": "Lorem Ipsum",
                },
                Object {
                  "body": "Lorem Ipsum",
                  "email": "lorem@ipsum",
                  "name": "Lorem Ipsum",
                },
              ],
            },
            "_currentValue2": Object {
              "comments": Array [
                Object {
                  "body": "Lorem Ipsum",
                  "email": "lorem@ipsum",
                  "name": "Lorem Ipsum",
                },
                Object {
                  "body": "Lorem Ipsum",
                  "email": "lorem@ipsum",
                  "name": "Lorem Ipsum",
                },
              ],
            },
            "_threadCount": 0,
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
